### PR TITLE
Better structure for the config file of panels

### PR DESCRIPTION
**Pull request by Vane11ope**
Now the config file of panels is like .local/share/radare/.r2panels but this PR modifies that to .local/share/radare/.r2panels/layout1 

it does not save tabs for now, but the feature also is coming shortly.

### CODE REVIEW

File: libr/core/panels.c [hunk 1]
```diff
@@ -4362,28 +4366,14 @@ void __update_menu_contents(RCore *core, RPanelsMenu *menu, RPanelsMenuItem *par
...
-		if (R_STR_ISEMPTY (p_cfg)) {
-			break;
-		}
-		tmp_cfg = __parse_panels_config (p_cfg, strlen (p_cfg));
-		name = sdb_json_get_str (tmp_cfg, "Name");
-		if (!name) {
-			break;
+	struct dirent *d;
+	DIR *dir_path = opendir (__get_panels_config_dir_path ());
+	while ((d = readdir (dir_path))) {
```
Comment (radare) [inline libr/core/panels.c]: check for null

File: libr/core/panels.c [hunk 2]
```diff
@@ -4362,28 +4366,14 @@ void __update_menu_contents(RCore *core, RPanelsMenu *menu, RPanelsMenuItem *par
...
-	for (i = 0; i < count; i++) {
-		if (R_STR_ISEMPTY (p_cfg)) {
-			break;
-		}
-		tmp_cfg = __parse_panels_config (p_cfg, strlen (p_cfg));
-		name = sdb_json_get_str (tmp_cfg, "Name");
-		if (!name) {
-			break;
+	struct dirent *d;
+	DIR *dir_path = opendir (__get_panels_config_dir_path ());
```
Comment (radare) [inline libr/core/panels.c]: i would say check for null in dirpath too

File: libr/core/panels.c [hunk 3]
```diff
@@ -5487,39 +5477,56 @@ void __restore_panel_pos(RPanel* panel) {
...
+		if (!strcmp (d->d_name, file)) {
+			ret = d->d_name;
+		}
 	}
-	char *newPath = r_str_home (configPath);
-	R_FREE (configPath);
-	return newPath;
+	closedir (dir);
+	if (!ret) {
+		return ret;
```
Comment (radare) [inline libr/core/panels.c]: do if (ret) { ... 
and put the char dir_path=, ret = ... after that. also the current code is leaking ret

File: libr/core/panels.c [hunk 4]
```diff
@@ -5528,17 +5535,15 @@ void r_save_panels_layout(RCore *core, const char *_name) {
 		pj_kn (pj, "h", panel->view->pos.h);
 		pj_end (pj);
 	}
-	FILE *file = r_sandbox_fopen (config_path, "ab");
+	FILE *file = r_sandbox_fopen (config_path, "w");
```
Comment (radare) [inline libr/core/panels.c]: +1

**Comment by Vane11ope** [comment]
done maybe

File: libr/core/panels.c [hunk 5]
```diff
@@ -5487,39 +5483,67 @@ void __restore_panel_pos(RPanel* panel) {
...
+	if (!dir) {
 		return NULL;
 	}
-	char *newPath = r_str_home (configPath);
-	R_FREE (configPath);
-	return newPath;
+	char *tmp = NULL;
+	struct dirent *d;
+	while ((d = readdir (dir))) {
+		if (!d) {
```
Comment (radare) [inline libr/core/panels.c]: This case will never happen. The while checks for it to leave the loop

File: libr/core/panels.c [hunk 6]
```diff
@@ -5487,39 +5483,67 @@ void __restore_panel_pos(RPanel* panel) {
 			panel->view->prevPos.w, panel->view->prevPos.h);
 }
 
-char *__get_panels_config_path() {
-	char *configPath = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
-	if (!configPath) {
+char *__get_panels_config_dir_path() {
+	char *config_path = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
+	char *new_config_path = r_str_home (config_path);
+	R_FREE (config_path);
```
Comment (radare) [inline libr/core/panels.c]: Use free() because this var is not used after this

File: libr/core/panels.c [hunk 7]
```diff
@@ -4361,29 +4365,21 @@ void __update_menu_contents(RCore *core, RPanelsMenu *menu, RPanelsMenuItem *par
...
 		return;
 	}
-	int count = r_str_split (panels_config, '\n');
-	char *name, *p_cfg = panels_config, *tmp_cfg;
-	for (i = 0; i < count; i++) {
-		if (R_STR_ISEMPTY (p_cfg)) {
-			break;
+	RCore *core = (RCore *)_core;
+	struct dirent *d;
+	while ((d = readdir (dir_path)) && d) {
```
Comment (radare) [inline libr/core/panels.c]: No need to check 3 times if d is not null on every iteration

File: libr/core/panels.c [hunk 8]
```diff
@@ -5487,39 +5483,67 @@ void __restore_panel_pos(RPanel* panel) {
 			panel->view->prevPos.w, panel->view->prevPos.h);
 }
 
-char *__get_panels_config_path() {
-	char *configPath = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
-	if (!configPath) {
+char *__get_panels_config_dir_path() {
+	char *config_path = r_str_new (R_JOIN_2_PATHS (R2_HOME_DATADIR, ".r2panels"));
```
Comment (radare) [inline libr/core/panels.c]: Lets be gentle and assume the heap cant fail?

**Pull request merged**